src/Routing/Middleware/RoutingMiddleware.php

```diff
@@ -45,10 +45,7 @@ public function __invoke(ServerRequestInterface $request, ResponseInterface $res
                 }
                 $request = $request->withAttribute(
                     'params',
-                    Router::parse(
-                        $request->getUri()->getPath(),
-                        $request->getMethod()
-                    )
+                    Router::parseRequest($request)
```

Comment:
I wanted to keep the deprecations small, so only `Router` and `RouteCollection` _needed_ new methods.